server/app/src/main/java/com/google/heartrate/wearos/app/bluetooth/BluetoothUtils.java

```diff
@@ -0,0 +1,120 @@
+package com.google.heartrate.wearos.app.bluetooth;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothGattServer;
+import android.bluetooth.BluetoothGattServerCallback;
+import android.bluetooth.BluetoothManager;
+import android.bluetooth.le.BluetoothLeAdvertiser;
+import android.content.Context;
+import android.content.pm.PackageManager;
+
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import static android.content.Context.BLUETOOTH_SERVICE;
+
+/**
+ * Provides bluetooth related objects and checks.
+ */
+public class BluetoothUtils {
+    private static final String TAG = BluetoothUtils.class.getSimpleName();
+
+    /**
+     * Assert that bluetooth is supported in current device.
+     *
+     * @param context application context
+     * @throws GattException if bluetooth is not supported
+     */
+    public static void assertBluetoothIsSupported(Context context) throws GattException {
+        BluetoothManager bluetoothManager = (BluetoothManager) context.getSystemService(BLUETOOTH_SERVICE);
+
+        if (bluetoothManager == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothManager found.");
+        }
+
+        BluetoothAdapter bluetoothAdapter = bluetoothManager.getAdapter();
+
+        if (bluetoothAdapter == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothAdapter found.");
+        }
+
+        if (!bluetoothAdapter.isEnabled()) {
+            throw new GattException("Bluetooth adapter is currently disabled.");
+        }
+
+        if (!context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)) {
+            throw new GattException("Bluetooth LE  is not supported. No FEATURE_BLUETOOTH_LE found.");
```

Comment:
typo: double space

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/4/commits/46e4bfbf8964c873edb648e1d78bd22b61070778)